utils/src/main/java/com/cloud/utils/net/cidr/CIDR.java

```diff
@@ -0,0 +1,98 @@
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+package com.cloud.utils.net.cidr;
+
+import java.net.InetAddress;
+import com.cloud.utils.Nullable;
+
+public interface CIDR extends Comparable<CIDR>, Nullable {
+
+    final static CIDR NULL = new NullCIDR();
+    final static int MAX_CIDR = 32;
+
+    public InetAddress getBaseAddress();
+
+    public int getMask();
+
+    public boolean contains(InetAddress inetAddress);
+
+    public boolean contains(CIDR cidr);
+
+    public boolean overlaps(CIDR cidr);
+
+    public String toString();
+
+    public boolean equals(Object o);
+
+    public int hashCode();
```

Comment:
It is unnecessary to declare ``toString``, ``equals``, and ``hashCode`` on an interface since all classes are ancestors of ``Object``.  Therefore, these method declarations can be removed.